src/Preprocessor.zig

```diff
@@ -365,6 +365,72 @@ fn preprocessExtra(pp: *Preprocessor, source: Source) MacroError!Token {
                             else => unreachable,
                         }
                     },
+                    .keyword_elifdef => {
+                        if (if_level == 0) {
+                            try pp.err(directive, .elifdef_without_if);
+                            if_level += 1;
+                            if_kind.set(if_level, until_else);
+                        } else if (if_level == 1) {
+                            guard_name = null;
+                        }
+                        switch (if_kind.get(if_level)) {
+                            until_else => {
+                                const macro_name = (try pp.expectMacroName(&tokenizer)) orelse continue;
```

Comment:
I think a missing macro name should be considered `false` similarly to a missing expression with `#if`.

Reply:
I have tested it with clang and you are right, i have also added a test for checking the missing macro name in the commit ef844b664b3ff4f6a025ae415669c94f2665ac22

Reply:
Clang generate this output:
```cpp
test/cases/#elifdefc2x_error.c:7:9: error: macro name missing
#elifdef
        ^
# 10 "test/cases/#elifdefc2x_error.c"
inttest/cases/#elifdefc2x_error.c:16:10: error: macro name missing
#elifndef
         ^

# 19 "test/cases/#elifdefc2x_error.c"
int
2 errors generated.
```
And Gcc genete this:
```cpp
# 10 "test/cases/#elifdefc2x_error.c"
int
# 19 "test/cases/#elifdefc2x_error.c"
int
```
I think the behaviour of gcc looks more correct, i'm going to commit to the pull request to reflect this.

Reply:
If there are not more issues, I think that with commit e1567309b0af3e27dad70a22e083aecedc5c3183 it should be ready to merge.

Reply:
It looks like they have the same output with clang missing a newline after the `int`.